### PR TITLE
Add Pinecone Writer Docs

### PR DESCRIPTION
**Pull request by karansampath**
Adds docs for the Pinecone docs and adjusts the notebook to ensure it works properly

### CODE REVIEW

File: docs/source/data_ingestion_and_preparation/connectors/pinecone.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Pinecone
+
+[Pinecone](https://www.pinecone.io/) is a vector database that supports serverless queries. Its optimization and focus on vector embedding and retrieval, and can be a good option to host large amounts of vector data.
+
+## Configuration for Pinecone
+
+*Please look at the [Pinecone API documentation](https://docs.pinecone.io/home) for in-depth background on the following. We specify the essential portions for creating a new Vector DB outside of Sycamore below.*
+
+Pinecone exist solely in its cloud infrastructure hosted via AWS. To set up a new Pinecone GRPC client connection in Python, generate a new [Pinecone API key](https://www.app.pinecone.io/), install the *pinecone* python package and run the following code:
```
Comment (HenryL27) [inline docs/source/data_ingestion_and_preparation/connectors/pinecone.md:9]: nit: exist -> exists
nit: you can also do pinecone in GCS I think

File: docs/source/data_ingestion_and_preparation/connectors/pinecone.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+- index_name: Name of the pinecone index to ingest into, is a required parameter.
+- index_spec: Cloud parameters needed by pinecone to create your index. See https://docs.pinecone.io/guides/indexes/create-an-index for additional information. Defaults to None, which assumes the index already exists, and will not modify an existing index if provided
+- namespace: Namespace withing the pinecone index to ingest into. See https://docs.pinecone.io/guides/indexes/use-namespaces for additional information. Defaults to "", which is the default namespace
+- dimensions: Dimensionality of dense vectors in your index. Defaults to None, which assumes the index already exists, and will not modify an existing index if provided.
+- distance_metric: Distance metric used for nearest-neighbor search in your index. Defaults to "cosine", but will not modify an already-existing index
+- api_key: Pinecone service API Key. Defaults to None (will use the environment variable PINECONE_API_KEY).
+- kwargs: Arguments to pass to the underlying execution engine
+
+To use the writer, call write at the end of a Sycamore pipeline as done below:
```
Comment (HenryL27) [inline docs/source/data_ingestion_and_preparation/connectors/pinecone.md:30]: example here or ref to example in api docs?

Comment (karansampath) [reply docs/source/data_ingestion_and_preparation/connectors/pinecone.md:30]: Have referenced below

**Review by jonfritz — APPROVED** [review]
LGTM!

**Pull request merged**